src/mimefactory.rs

```diff
@@ -1314,6 +1325,17 @@ impl<'a> MimeFactory<'a> {
             let json = self.msg.param.get(Param::Arg).unwrap_or_default();
             parts.push(context.build_status_update_part(json));
         } else if self.msg.viewtype == Viewtype::Webxdc {
+            let topic = create_random_topic();
+            let peer = Box::pin(context.get_or_generate_iroh_keypair())
```

Comment:
Isn't this expensive if the app doesn't use iroh channels?

Reply:
This only happens only for webxdc messages which you don't send too often. Sending the 30kb should deminish the performance implication of two SQL-queries and creating some random bytes.